### ISSUE TITLE
Config search regression introduced in 2.3.0

### ISSUE THREAD

**Issue by hudochenkov** [TheSavior/ESLint-Formatter#52]
I have a monorepo project (inside project multiple sub-project each with own package.json). ESLint is set up in the project root directory, and not in sub-directories with own package.json. With [this change](https://github.com/TheSavior/ESLint-Formatter/commit/888f11908b7d49d243f8960f297ea72a19a9a5db) introduced in 2.3.0 ESLint-Formatter looking in the closest package.json, but if there is no ESLint, it won't look into project root, and go straight to globally installed ESLint.

```
Using node.js path on 'osx': /usr/local/bin/node
Using eslint path on 'osx': /usr/local/bin/eslint
Unexpected error(<class 'Exception'>): Error: module.js:471
    throw err;
    ^

Error: Cannot find module '/usr/local/bin/eslint'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.runMain (module.js:604:10)
    at run (bootstrap_node.js:389:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:504:3

error: Error: module.js:471
    throw err;
    ^

Error: Cannot find module '/usr/local/bin/eslint'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.runMain (module.js:604:10)
    at run (bootstrap_node.js:389:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:504:3
```

**Comment by hudochenkov** [comment]
Currently, I set up project-specific settings override as a workaround for this issue:

```json
{
    "folders":
    [
        {
            "path": "."
        }
    ],
    "settings": {
        "ESLint-Formatter": {
            "eslint_path": {
                "osx": "/Users/aleks/project/node_modules/.bin/eslint"
            }
        }
    }
}
```

**Comment by TheSavior** [comment]
@ramiel, how should this be handled?

**Comment by ramiel** [comment]
I'll look at this tomorrow. We should stop at the project root instead of the first package.json probably.

**Comment by ramiel** [comment]
The problem can be a bit complex. At the moment we go up, starting from the current directory, and we stop at the first package.json we find. Then we suppose that is the point were eslint is installed. Setup like the one from @hudochenkov show that this is not always the case. We have so two possibilities.

- Look if the package.json reference eslint among its dependencies
- Look if a .eslint.* file is present

The latter is not affordable because eslint allow to have more than one file and do not enforce you to put `"root": true` in the top one.    
Reading the content of the package.json file instead is heavy, considering that the plugin has no cache system and that this operation is performed everytime a file if linted (easily at every save).
It looks to me that using a per-project configuration for cases like this is the most viable solution.     
@TheSavior have you any other idea?

**Comment by ramiel** [comment]
Another idea is to look for the first path where is present `node_modules/eslint`. This rely on the structure of the node_modules folder itself, which should be reliable until npm change it again or another fancy package manager is used. But it's fast. I'll propose a PR and we continue to discuss there
